imbi/endpoints/project_dependencies.py

```diff
@@ -71,6 +73,75 @@ async def get(self, *args, **kwargs):
         else:
             await super().get(*args, **kwargs)
 
+    async def post(self, *args, **kwargs):
+        values = self.get_request_body()
+        values['project_id'] = kwargs['project_id']
+        values['username'] = self._current_user.username
+        dependent_project = await models.project(values['project_id'],
+                                                 self.application)
+        selected_automations = await automations.retrieve_automations(
+            self.application, values.get('automations', []),
+            dependent_project.project_type.id)
+
+        result = await self.postgres_execute(self.POST_SQL, values,
+                                             f'post-{self.NAME}')
+        if not result.row_count:
+            raise errors.DatabaseError('Failed to create project dependency',
+                                       title='Failed to create record')
+
+        dependency = await models.project_dependency(
+            result.row['project_id'], result.row['dependency_id'],
+            self.application)
```

Comment:
You could avoid the additional query by having the `RETURNING` clause in `POST_SQL` return what you need to create a `Dependency`

Reply:
I can. I originally kept it in because we have a pattern of doing the query on every POST request after the new row is `INSERT`ed. Although, tbh I'd rather we not and just leverage `RETURNING`

Reply:
That pattern is in there somewhere ... IIRC the implementation of `post()` in the base class invokes `_post()` and then `_get()` or something like that. I have to relearn which class attributes need to be set for that to work _every time that I touch it_.